packages/compass-collection/src/stores/collection.ts

```diff
@@ -0,0 +1,113 @@
+import type AppRegistry from 'hadron-app-registry';
+import type { DataService } from 'mongodb-data-service';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import reducer, {
+  collectionStatsFetched,
+  selectTab,
+} from '../modules/collection-tab';
+import type Collection from 'mongodb-collection-model';
+import toNs from 'mongodb-ns';
+import type { MongoDBInstance } from 'mongodb-instance-model';
+import type { CollectionMetadata } from 'mongodb-collection-model';
+
+export function configureStore(
+  options: {
+    dataService: DataService;
+    globalAppRegistry: AppRegistry;
+    localAppRegistry: AppRegistry;
+    query?: unknown;
+    aggregation?: unknown;
+    editViewName?: string;
+  } & CollectionMetadata
+) {
+  const {
+    dataService,
+    globalAppRegistry,
+    localAppRegistry,
+    query,
+    aggregation,
+    editViewName,
+    ...collectionMetadata
+  } = options;
+
+  const instance = (
+    globalAppRegistry.getStore('App.InstanceStore') as
+      | {
+          getInstance(): MongoDBInstance;
+        }
+      | undefined
+  )?.getInstance();
+
+  if (!instance) {
+    throw new Error('Expected to get instance from App.InstanceStore');
+  }
+
+  const configureFieldStore = globalAppRegistry.getStore('Field.Store') as (
+    ...args: any
+  ) => void | undefined; // our handcrafted d.ts file doesn't match the actual code
+
+  configureFieldStore?.({
+    localAppRegistry: localAppRegistry,
+    globalAppRegistry: globalAppRegistry,
+    namespace: collectionMetadata.namespace,
+    serverVersion: instance.build.version,
+  });
+
+  const { database, collection } = toNs(collectionMetadata.namespace);
+
+  const store = createStore(
+    reducer,
+    {
+      metadata: {
+        ...collectionMetadata,
+        // It's safe to just read this once here on store creation because
+        // instance is fully instantiated and instance details are fetched at
+        // the point when tab is created
```

Comment:
Maybe worth an additional note here that "... and these instance props will not change during the connection lifecycle"

Reply:
I mean … technically the server version could change, right? You can do rolling server version updates on replica sets without downtime.

In any case, I think it might be good to clarify what the thing/actual code is that ensures the guarantee made here in the comment.

Reply:
Yeah, that's a good point, and instance model indeed never accounted for that, I'll add a more thorough description here and in the instance model

Reply:
Updated comment in https://github.com/mongodb-js/compass/pull/4858/commits/f44a185908b78c334041a47587de1957d77a264e